File: 07/07/e.py
import math

t = int(input())

for _ in range(t):
    n = int(input())
    fl = 0
    for k in range(2, 200):
        total = 1 + k + k * k
        cur = k * k * k
        while total < n:
            total += cur
            cur *= k
        if total == n:
            print("YES")
            fl = 1
            break
    if fl == 0:
        k = int(-1 + math.sqrt(4 * n - 3)) // 2
        if 1 + k + k * k == n and k > 1:
            print("YES")
        else:
            print("NO")

File: 07/07/e.cpp
#include <bits/stdc++.h>
using namespace std;
#define int long long

const int mx = 1000'000'000'000'000'000;

signed main() {
    set<int> st;
    for(int k = 2; k <= 1000000; ++k) {
        int total = 1 + k + k * k;
        int cur = k * k;
        for(int m = 3; m <= 63; ++m) {
            if(cur > mx / k) break;
            cur *= k;
            total += cur;
            st.insert(total);
        }
    }

    int t;
    cin >> t;
    while(t--) {
        int n;
        cin >> n;
        if(st.count(n)) {
            cout << "YES\n";
        } else {
            int k = ((int)sqrt(4 * n - 3) - 1) / 2;
            if(k > 1 && 1 + k + k * k == n) {
                cout << "YES\n"; 
            } else {
                cout << "NO\n";
            }
        }
    }
}